packages/components/src/dropdown-menu/index.tsx

```diff
@@ -62,13 +139,16 @@ function DropdownMenu( dropdownMenuProps ) {
 	}
 
 	// Normalize controls to nested array of objects (sets of controls)
-	let controlSets;
+	let controlSets: NonNullable< typeof controls >[];
+
 	if ( controls?.length ) {
-		controlSets = controls;
-		if ( ! Array.isArray( controlSets[ 0 ] ) ) {
-			controlSets = [ controlSets ];
+		if ( ! Array.isArray( controls[ 0 ] ) ) {
+			controlSets = [ controls ];
+		} else {
+			controlSets = controls as unknown as typeof controlSets;
```

Comment:
I don't like this double assertion, but I didn't see another way to convince TS that `controls` is definitely a nested array at this point.